From: Xose Vazquez Perez <xose.vazquez@gmail.com>
Date: Tue, 24 Jun 2025 21:09:25 +0200
Subject: scsi: qla2xxx: Remove firmware URL
Patch-mainline: v6.17-rc1
Git-commit: b152f199fa43ef96ecb29cefafa1e9b0e02b6f3d
References: git-fixes

The historic QLogic firmware URL redirects to a Marvell page that only
provides drivers.

Refer to linux-firmware instead.

Cc: Nilesh Javali <njavali@marvell.com>
Cc: James E.J. Bottomley <James.Bottomley@HansenPartnership.com>
Cc: Martin K. Petersen <martin.petersen@oracle.com>
Cc: QLOGIC ML <GR-QLogic-Storage-Upstream@marvell.com>
Cc: LINUX SCSI ML <linux-scsi@vger.kernel.org>
Signed-off-by: Xose Vazquez Perez <xose.vazquez@gmail.com>
Link: https://lore.kernel.org/r/20250624190926.115009-1-xose.vazquez@gmail.com
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Daniel Wagner <dwagner@suse.de>
---
 drivers/scsi/qla2xxx/Kconfig    |    6 +-----
 drivers/scsi/qla2xxx/qla_init.c |    4 ----
 2 files changed, 1 insertion(+), 9 deletions(-)

--- a/drivers/scsi/qla2xxx/Kconfig
+++ b/drivers/scsi/qla2xxx/Kconfig
@@ -25,11 +25,7 @@ config SCSI_QLA_FC
 	  Upon request, the driver caches the firmware image until
 	  the driver is unloaded.
 
-	  Firmware images can be retrieved from:
-
-	        http://ldriver.qlogic.com/firmware/
-
-	  They are also included in the linux-firmware tree as well.
+	  Firmware images are included in the linux-firmware tree.
 
 config TCM_QLA2XXX
 	tristate "TCM_QLA2XXX fabric module for QLogic 24xx+ series target mode HBAs"
--- a/drivers/scsi/qla2xxx/qla_init.c
+++ b/drivers/scsi/qla2xxx/qla_init.c
@@ -8593,8 +8593,6 @@ qla24xx_load_risc_flash(scsi_qla_host_t
 	return QLA_SUCCESS;
 }
 
-#define QLA_FW_URL "http://ldriver.qlogic.com/firmware/"
-
 int
 qla2x00_load_risc(scsi_qla_host_t *vha, uint32_t *srisc_addr)
 {
@@ -8612,8 +8610,6 @@ qla2x00_load_risc(scsi_qla_host_t *vha,
 	if (!blob) {
 		ql_log(ql_log_info, vha, 0x0083,
 		    "Firmware image unavailable.\n");
-		ql_log(ql_log_info, vha, 0x0084,
-		    "Firmware images can be retrieved from: "QLA_FW_URL ".\n");
 		return QLA_FUNCTION_FAILED;
 	}
 
